rules-tests/Restoration/Rector/ClassLike/UpdateFileNameByClassNameFileSystemRector/UpdateFileNameByClassNameFileSystemRectorTest.php

```diff
@@ -17,13 +15,7 @@ final class UpdateFileNameByClassNameFileSystemRectorTest extends AbstractRector
      */
     public function test(SmartFileInfo $smartFileInfo): void
     {
-        $this->doTestFileInfo($smartFileInfo);
-
-        $expectedAddedFileWithContent = new AddedFileWithContent(
-            $this->originalTempFileInfo->getRealPathDirectory() . '/SkipDifferentClassName.php',
-            FileSystem::read(__DIR__ . '/Fixture/skip_different_class_name.php.inc')
-        );
-        $this->assertFileWasAdded($expectedAddedFileWithContent);
+        $this->doTestFileInfoWithAdditionalChanges($smartFileInfo);
```

Comment:
How does this exactly work? Looks magic

Reply:
It explodes fixture content to 3 parts - from part 1 and part 2 new file is created and used in original method `$this->doTestFileInfo()` - it tests changes in processed php file

3rd part is again exploded and creates triplets: filename, original content, new content and then they are tested against added files ($addedFilesWithContent = $this->removedAndAddedFilesCollector->getAddedFilesWithContent();)

Reply:
> Looks magic

is it a good or bad? :)